m3/src/main/java/com/uber/m3/tally/m3/thrift/TUdpTransport.java

```diff
@@ -50,6 +50,8 @@
     protected final SocketAddress socketAddress;
     protected final DatagramSocket socket;
 
+    protected volatile boolean open;
```

Comment:
probably leave some documentation here on why we are using a boolean to track the state of the socket given that we can use the Datagram Socket state as well. 

Also, would the socket close down without an explicit close?

Reply:
SG, will do. The state `isOpen` is reflecting could only be changed upon `close` invocation, so we're preserving the semantic here.